### PR TITLE
enh(notion) change how we process dbs

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- Remove all the inline database logic (which was broken anyway...). This may save quite a bit of requests to notion 🤔. We continue to add child databases we don't know about to "resources to check"
- Insert inline placeholders (`Child Database ${DB_TITLE}`) for databases that are children of a block
- upsert databases to semantic store as CSV files (using header as prefix)
- upsert pages that don't have a body if one of their properies has over 256 characters

## Risk

Notion sync could get stuck, but easy to rollback

## Deploy Plan

Requires a queue bump + restart of workflows

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 1]
```diff
@@ -2373,10 +2344,46 @@ export async function upsertDatabaseStructuredDataFromCache({
...
+    databaseId,
+    new Set<string>(),
+    runTimestamp.toString()
+  );
+  localLogger.info("Upserting Notion Database as Document.");
+  await upsertToDatasource({
+    dataSourceConfig,
+    documentId: `notion-database-${databaseId}`,
+    documentContent: {
+      prefix: `${databaseName}\ncsvHeader`,
```
Comment (spolu) [inline connectors/src/connectors/notion/temporal/activities.ts:2365]: csvHeader?

**Review by spolu — COMMENTED** [review]
Looks great but the prefix logic seems not adequate? We want the headler here and not in the csvForDocument

File: connectors/src/connectors/notion/temporal/activities.ts (hunk 1 above)
Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts:2365]: So this is the header right ? What you don't want is DB title ?

happy to remove csv header from the body though

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts]: Ok I see it ! Fixed (removed DB name from prefix, did we want it ?)

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts]: used the renderPrefixSection with the defaults -- should I override to have more than 64 tokens for the header ?

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts]: Ok ended up doubling it.

**Review by spolu — APPROVED** [review]
LGTM

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 2]
```diff
@@ -2373,10 +2338,50 @@ export async function upsertDatabaseStructuredDataFromCache({
...
+      databaseId,
+      new Set<string>(),
+      runTimestamp.toString()
+    );
+    localLogger.info("Upserting Notion Database as Document.");
+    await upsertToDatasource({
+      dataSourceConfig,
+      documentId: `notion-database-${databaseId}`,
+      documentContent: {
+        prefix: csvHeader,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts]: This is outdated right?

**Pull request merged**